node/lib/util/submodule_config_util.js

```diff
@@ -407,7 +407,7 @@ exports.initSubmoduleAndRepo = co.wrap(function *(repoUrl,
     }
     if (null !== origin) {
         if (realUrl !== origin.url()) {
-            yield NodeGit.Remote.setUrl(result, "origin", realUrl);
+            NodeGit.Remote.setUrl(result, "origin", realUrl);
```

Comment:
We didn't find it earlier because users have to get their submodule in a weird state (probably, e.g., from https://github.com/twosigma/git-meta/issues/196) to hit this case.